pl_bolts/models/self_supervised/byol/byol_module.py

```diff
@@ -75,77 +74,70 @@ def __init__(
         max_epochs: int = 1000,
         base_encoder: Union[str, torch.nn.Module] = "resnet50",
         encoder_out_dim: int = 2048,
-        projector_hidden_size: int = 4096,
+        projector_hidden_dim: int = 4096,
         projector_out_dim: int = 256,
-        **kwargs
-    ):
-        """
-        Args:
-            datamodule: The datamodule
-            learning_rate: the learning rate
-            weight_decay: optimizer weight decay
-            input_height: image input height
-            batch_size: the batch size
-            num_workers: number of workers
-            warmup_epochs: num of epochs for scheduler warm up
-            max_epochs: max epochs for scheduler
-            base_encoder: the base encoder module or resnet name
-            encoder_out_dim: output dimension of base_encoder
-            projector_hidden_size: hidden layer size of projector MLP
-            projector_out_dim: output size of projector MLP
-        """
+        initial_tau: float = 0.996,
+        **kwargs: Any,
```

Comment:
Quite a few of these args is not used, can we get rid of them? (I cannot comment on them specifically since you didn't change the code around them)

Reply:
I am keeping `learning_rate`, `weight_decay`, `warmup_epochs`, and `max_epochs` since they are called from `self.hparams` in `configure_optimizers()`.

Reply:
I don't think that `max_epochs` is used in `self.hparams` (and even if it is, it is not really appropriate, since it's not model-specific, but trainer-specific argument)

Reply:
Is it not necessary for configuring the scheduler (LinearWarmupCosineAnnealingLR)?

Reply:
I feel like I'm overlooking too much lately :facepalm: Of course it is used in the scheduler and therefore very much appropriate in here.